core/src/com/marama/view/renderables/World.java

```diff
@@ -13,21 +14,25 @@
 import com.badlogic.gdx.math.Vector3;
 import com.badlogic.gdx.math.collision.Ray;
 import com.badlogic.gdx.utils.Array;
-import com.marama.view.entities.MBlock;
+import com.badlogic.gdx.utils.ObjectMap;
+import com.marama.view.entities.EntityManagerSingleton;
+import com.marama.view.entities.Maramafication;
+import com.marama.view.entities.exceptions.ModelNotFoundException;
 import com.marama.view.entities.instances.SelectableInstance;
 
 /**
  * The {@link World} is an {@link Environment} that is able to render 3D {@link ModelInstance}'s
  */
 public class World extends Environment implements Renderable {
     private boolean loading;
+
     private DirectionalLight directionalLight;
-    private Array<ModelInstance> modelInstances; // All models that are in the World.
     private PerspectiveCamera perspectiveCamera;
+    private ModelBatch modelBatch; // The unit that can render the modelInstances
     private CameraInputController cameraInputController;
-    private AssetManager assetManager;
-    private ModelBatch modelBatch;
-    private MBlock mBlock;
+
+    private EntityManagerSingleton entityManager; // The unit that can hold all the models
```

Comment:
Comment really doesn't explain anything.

Reply:
This comment is gold.